### PR TITLE
Feature/336 upgrade to veramo 2

### PR DESCRIPTION
**Pull request by theblockstalk**
1. upgraded ALL packages to latest
    - including using the latest `did-resolver`, `did-jwt` and `did-jwt-vc` packages from DIF repo, instead of the outdated `@tonomy/did-...` bundles https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/issues/336
    - using latest `antelope-did-resolver` after it was upgraded https://github.com/Tonomy-Foundation/antelope-did-resolver/pull/15
    - removed `jsdom` and `jsdom-url` package as not needed in latest jest. Added jsdom dependencies in `./test/test.setup.ts` and removed jest custom environment
2. Jest runs all tests using ESM under
    - typeorm, uuid and ws packages could not be bundled fully with ESM so exceptions were made in `jest.config.ts`
    - `yarn run cli` also using ESM bundles
3. Added new `veramo.ts` and unit tests. to ensure we are ready for next tasks that depended on this upgrade https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/issues/337 to be ready for https://github.com/Tonomy-Foundation/Tonomy-ID/issues/758
4. removed `did-jwk.ts` and upgraded to use `did-key.ts` instead. this solves https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/issues/316
5. removed the `API` and `utils` builds and conditional exports
6. removed unused `base64.ts` helper functions and unit tests
7. added new CI pipeline to check for latest packages - but it does not block PR merges, just there as a warning to remind developers
8. upgraded `eslint` to use the new configuration file and conventions. Also removing and updating config dependency packages
9. upgraded to yarn 4.3.1
10. installed [debug](https://www.npmjs.com/package/debug) for debug logging
11. fixed issue with `ExternalUser.getIssuer()` using the wrong `keyManager`
12. fixed all tests to work again

### CODE REVIEW

**Comment by theblockstalk** [comment]
Updated the resolver to use latest libraries and noble curves: https://github.com/Tonomy-Foundation/antelope-did-resolver/pull/15

**Comment by theblockstalk** [comment]
TODO:
- update to latest `did-antelope-resolver@0.9.0`
- use createIssuer, and creatSigner function fro `did-antelope-resolver`

File: src/sdk/util/ssi/veramo.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    return verifiableCredential;
+}
+
+async function verifyCredential(agent: AgentType, verifiableCredential: VerifiableCredential) {
+    const result = await agent.verifyCredential({ credential: verifiableCredential });
+
+    debug(`Credential verified`, result.verified);
+}
+
+export async function veramo2() {
```
Comment (sadiabbasi) [inline src/sdk/util/ssi/veramo.ts:145]: veramo and veramo2 name.. may be we can update this name?

**Pull request merged**